### PR TITLE
feature(client&service):

### PR DESCRIPTION
**Pull request by MarvinMiao**
#90 
1. support mongodb
2. add getLatestVersionByKey api in service

### CODE REVIEW

File: flink-ai-flow/lib/notification_service/java/src/test/java/com/aiflow/notification/client/NotificationClientTest.java [hunk 1]
```diff
@@ -109,4 +109,14 @@ public void process(List<Event> events) {
 		assertEquals(3, ii[0].intValue());
 	}
 
-}
\ No newline at end of file
+	@Test
+	public void getLatestVersion() throws Exception {
+		for(int i = 0; i <3; i++) {
```
Comment (SteNicholas) [inline flink-ai-flow/lib/notification_service/java/src/test/java/com/aiflow/notification/client/NotificationClientTest.java:114]: This should add space to follow the code style.

File: flink-ai-flow/lib/notification_service/notification_service/base_notification.py [hunk 2]
```diff
@@ -116,3 +181,11 @@ def stop_listen_events(self):
         :return:
         """
         pass
+    
+    @abc.abstractmethod
+    def get_latest_version(self, key: str = None):
+        """
+        get latest event's version by key
+        :param key: event's key
```
Comment (SteNicholas) [inline flink-ai-flow/lib/notification_service/notification_service/base_notification.py:189]: The description of parameter `key` could be `Key of notification for listening.`.

File: flink-ai-flow/lib/notification_service/notification_service/base_notification.py [hunk 3]
```diff
@@ -116,3 +181,11 @@ def stop_listen_events(self):
         :return:
         """
         pass
+    
+    @abc.abstractmethod
+    def get_latest_version(self, key: str = None):
+        """
+        get latest event's version by key
```
Comment (SteNicholas) [inline flink-ai-flow/lib/notification_service/notification_service/base_notification.py:188]: Add `.` at the end of this comment.

File: flink-ai-flow/lib/notification_service/notification_service/client.py [hunk 4]
```diff
@@ -250,3 +251,18 @@ def stop_listen_events(self):
                 del self.threads[ALL_EVENTS_KEY]
         finally:
             self.lock.release()
+
+    def get_latest_version(self, key: str = None):
+        """
+        get latest event's version by key
```
Comment (SteNicholas) [inline flink-ai-flow/lib/notification_service/notification_service/client.py:257]: The same to above.

File: flink-ai-flow/lib/notification_service/notification_service/client.py [hunk 5]
```diff
@@ -250,3 +251,18 @@ def stop_listen_events(self):
...
+        :param key: event's key
```
Comment (SteNicholas) [inline flink-ai-flow/lib/notification_service/notification_service/client.py:258]: The same to above.

File: flink-ai-flow/lib/notification_service/notification_service/event_storage.py [hunk 6]
```diff
@@ -72,9 +74,75 @@ def list_all_events_from_id(self, id: int):
...
+            "host": kwargs.get("host"),
+            "port": kwargs.get("port"),
+            "username": kwargs.get("username"),
+            "password": kwargs.get("password"),
+            "db": kwargs.get("db"),
+        }
+        # If ignore the authSource, the authentication will be failed.
+        conn_uri = "mongodb://{username}:{password}@{host}:{port}/{db}?authSource=admin".format(**db_conf)
+        db_conf["host"] = conn_uri
+        print("#####mongo db conf: {}".format(str(db_conf)))
```
Comment (SteNicholas) [inline flink-ai-flow/lib/notification_service/notification_service/event_storage.py:108]: This print could be removed or recommend to use logging to log the MongoDB configuration.

**Review by SteNicholas — CHANGES_REQUESTED** [review]

File: flink-ai-flow/lib/notification_service/notification_service/event_storage.py [hunk 7]
```diff
@@ -73,8 +75,73 @@ def list_all_events_from_id(self, id: int):
...
+                res.append(event.version)
+        res.sort()
+        return res[0] if res else 0
+
     def clean_up(self):
         self.store.clear()
         self.kv.clear()
         self.max_id = 0
 
+
```
Comment (wuchaochen) [inline flink-ai-flow/lib/notification_service/notification_service/event_storage.py:91]: add a new python file

**Review by wuchaochen — APPROVED** [review]

**Pull request merged**